madgraph/various/process_checks.py

```diff
@@ -1267,12 +1267,8 @@ def time_matrix_element(self, matrix_element, reusing = False,
         dir_name=res_timings['dir_path']
 
         def check_disk_usage(path):
-            return subprocess.Popen("du -shc -L "+str(path), \
-                stdout=subprocess.PIPE, shell=True).communicate()[0].decode(errors='ignore').split()[-2]
-            # The above is compatible with python 2.6, not the neater version below
-            # -> need to check if need .decode for python3.7
-            #return subprocess.check_output(["du -shc %s"%path],shell=True).\
-            #                                                         split()[-2]
+            return subprocess.check_output([f"du -shc {path}"], shell=True).\
```

Comment:
Use python f-strings here as they were added in Python 3.6 and this is Python 3.7+ code.

Reply:
RE: the comment

```
# -> need to check if need .decode for python3.7
```

this can be easily dealt with by using the standard `encoding="utf8"` to get strings back instead of bytestrings.

Example:

```console
$ docker run --rm -ti python:3.10 /bin/bash
root@457ba718d9b0:/# cd home/
root@457ba718d9b0:/home# du -shc $PWD
4.0K	/home
4.0K	total
root@457ba718d9b0:/home# python
Python 3.10.10 (main, Feb 11 2023, 02:40:33) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> import subprocess
>>> path = Path.cwd()
>>> subprocess.check_output([f"du -shc {path}"], shell=True).split()[-2]
b'4.0K'
>>> subprocess.check_output([f"du -shc {path}"], shell=True, encoding="utf8").split()[-2]
'4.0K'
>>>
```